From a4ad0d30d1057bfe5ae9abede74d2ebb2df0c51d Mon Sep 17 00:00:00 2001
From: Allan Stephens <allan.stephens@windriver.com>
Date: Fri, 5 Nov 2010 10:02:15 -0400
Subject: [PATCH 04/32] tipc: Eliminate warnings associated with node signature and flag values

commit 26b58e1759edc50307807bc8553f0c3003927d77 from
git://tipc.cslab.ericsson.net/pub/git/people/allan/tipc.git

Increases the size of TIPC's node structure fields for signature and
flag values to prevent warnings about potential loss of high order
significant bits when these values are updated. (Note: The fields are
already big enough to hold these values correctly, but the compiler
doesn't know this.)

Signed-off-by: Allan Stephens <allan.stephens@windriver.com>
Integrated-by: Howard Xu <hao.xu@windriver.com>
---
 net/tipc/tipc_node.h |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/net/tipc/tipc_node.h b/net/tipc/tipc_node.h
index 5075254..0b8bba2 100644
--- a/net/tipc/tipc_node.h
+++ b/net/tipc/tipc_node.h
@@ -2,7 +2,7 @@
  * net/tipc/tipc_node.h: Include file for TIPC node management routines
  * 
  * Copyright (c) 2000-2006, Ericsson AB
- * Copyright (c) 2005-2008, Wind River Systems
+ * Copyright (c) 2005-2008, 2010, Wind River Systems
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -76,8 +76,8 @@ struct tipc_node {
 	int working_links;
 	int permit_changeover;
 	int cleanup_required;
-	u16 signature;
-	u16 flags;
+	unsigned int signature;
+	unsigned int flags;
 	struct {
 		char supportable;
 		char supported;
-- 
1.7.0

